src/ggl-setup.py

```diff
@@ -0,0 +1,524 @@
+                   group: str, user: str) -> bool:
+
+    try:
+        iot_data_endpoint = iot_client.describe_endpoint(
+            endpointType='iot:Data-ATS')['endpointAddress']
+        iot_cred_endpoint = iot_client.describe_endpoint(
+            endpointType='iot:CredentialProvider')['endpointAddress']
+
+        with open(file_path, 'r') as file:
+            data = yaml.safe_load(file)
+
+        data['system']['thingName'] = thing_name
+        data['system']['privateKeyPath'] = PRIVATE_PATH
+        data['system']['certificateFilePath'] = DEVICE_PATH
+        data['system']['rootCaPath'] = CA_PATH
+        data['services']['aws.greengrass.NucleusLite']['configuration'][
+            'iotCredEndpoint'] = iot_cred_endpoint
+        data['services']['aws.greengrass.NucleusLite']['configuration'][
+            'iotDataEndpoint'] = iot_data_endpoint
+        data['services']['aws.greengrass.NucleusLite']['configuration'][
+            'runWithDefault']['posixUser'] = group + ':' + user
+        data['services']['aws.greengrass.NucleusLite']['configuration'][
+            'iotRoleAlias'] = "ggl-uat-role-alias"
+        print(data)
+
+        with open(file_path, 'w') as file:
+            yaml.dump(data, file, default_flow_style=False)
+
+        print(f"Successfully modified the config file")
+        return True
+
+    except Exception as e:
+        print(f"Error when modifying config.yaml: {str(e)}")
+        return False
+
+
+def _stop_and_disable_services():
+    commands = [
+        ['sudo', 'systemctl', 'stop', 'greengrass-lite.target'],
+        ['sudo', 'systemctl', 'disable', 'greengrass-lite.target'],
+        ['sudo', 'systemctl', 'disable', 'ggl.aws_iot_tes.socket'],
+        ['sudo', 'systemctl', 'disable', 'ggl.aws_iot_mqtt.socket'],
+        ['sudo', 'systemctl', 'disable', 'ggl.gg_config.socket'],
+        ['sudo', 'systemctl', 'disable', 'ggl.gg_health.socket'],
+        ['sudo', 'systemctl', 'disable', 'ggl.gg_fleet_status.socket'],
+        ['sudo', 'systemctl', 'disable', 'ggl.gg_deployment.socket'],
+        ['sudo', 'systemctl', 'disable', 'ggl.gg_pubsub.socket'],
+        ['sudo', 'systemctl', 'disable', 'ggl.gg-ipc.socket.socket'],
+        ['sudo', 'systemctl', 'disable', 'ggl.core.ggconfigd.service'],
+        ['sudo', 'systemctl', 'disable', 'ggl.core.iotcored.service'],
+        ['sudo', 'systemctl', 'disable', 'ggl.core.tesd.service'],
+        ['sudo', 'systemctl', 'disable', 'ggl.core.ggdeploymentd.service'],
+        ['sudo', 'systemctl', 'disable', 'ggl.core.gg-fleet-statusd.service'],
+        ['sudo', 'systemctl', 'disable', 'ggl.core.ggpubsubd.service'],
+        ['sudo', 'systemctl', 'disable', 'ggl.core.gghealthd.service'],
+        ['sudo', 'systemctl', 'disable', 'ggl.core.ggipcd.service'],
+        ['sudo', 'systemctl', 'daemon-reload'],
+        ['sudo', 'rm', '-rf', '/etc/systemd/system/ggl*']
+    ]
+
+    for cmd in commands:
+        try:
+            subprocess.run(cmd, check=True)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method is running with elevated privileges. Privilege elevation occurs when you access resources with an extended set of permissions. When unattended, these privileges might allow attackers to compromise the system. Do not use elevated privileges when they can be avoided.

[Learn more](https://cwe.mitre.org/data/definitions/269.html)